src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java

```diff
@@ -46,29 +53,38 @@
     private AblyHelpers.AppSpec appSpec;
     private AblyRealtime ablyClient;
 
-    @BeforeEach
-    public void setup() throws Exception {
-        appSpec = AblyHelpers.createTestApp();
+    @BeforeAll
+    public void prepTestEnvironment() throws Exception {
+        assertDoesNotThrow(() -> appSpec = AblyHelpers.createTestApp(), "Failed to create Ably client");
 
         connectCluster = new EmbeddedConnectCluster.Builder().build();
         connectCluster.start();
-        connectCluster.assertions().assertAtLeastNumWorkersAreUp(NUM_WORKERS, "Initial group of workers did not start in time.");
+        connectCluster.kafka().createTopic(DEFAULT_TOPIC);
 
         ablyClient = AblyHelpers.realtimeClient(appSpec);
+        final CountDownLatch latch = new CountDownLatch(1);
+        ablyClient.connection.on(connectionStateChange -> {
+            if (connectionStateChange.current == ConnectionState.connected) {
+                latch.countDown();
+            } else if (connectionStateChange.current == ConnectionState.failed) {
+                latch.countDown();
+                assertTrue(false, "Connection failed");
+            }
+        });
+        latch.await();
     }
```

Comment:
I'd extract this part to a separate method called something like `assertAblyIsConnected()` to make it clear what are we trying to do here :wink: 
```suggestion
        ablyClient = AblyHelpers.realtimeClient(appSpec);
        assertAblyIsConnected();
    }

    private void assertAblyIsConnected() {
        final CountDownLatch latch = new CountDownLatch(1);
        ablyClient.connection.on(connectionStateChange -> {
            if (connectionStateChange.current == ConnectionState.connected) {
                latch.countDown();
            } else if (connectionStateChange.current == ConnectionState.failed) {
                latch.countDown();
                assertTrue(false, "Connection failed");
            }
        });
        latch.await();
    }
```

Reply:
a6c110fd320ce329dedc8bbdc1956d6c750f7a83